### PR TITLE
Ensure script engine factories are loaded before rule engine starts

### PR DESCRIPTION
**Pull request by J-N-K**
Depends on https://github.com/openhab/openhab-core/pull/3275

This MUST NOT be merged before, otherwise the system hangs at startlevel 20.

Signed-off-by: Jan N. Klug <github@klug.nrw>

### CODE REVIEW

**Comment by J-N-K** [comment]
@kaikreuzer  I have rebased and fixed merged conflicts. Can you merge this one so that we can use it?

File: distributions/openhab/src/main/resources/runtime/services.cfg [hunk 1]
```diff
@@ -14,8 +14,9 @@ threadpool:safeCall=10
 threadpool:thingHandler=5
 
 # Start level definitions
-startlevel:20=dsl:items,managed:item,dsl:things,managed:thing,managed:itemchannellink,dsl:persist,managed:metadata
-startlevel:30=persistence:restore
+ startlevel:20=dsl:items,managed:item,dsl:things,managed:thing,managed:itemchannellink,dsl:persist,managed:metadata
```
Comment (kaikreuzer) [inline distributions/openhab/src/main/resources/runtime/services.cfg:17]: Did you add the space in front on purpose or by accident?

File: distributions/openhab/src/main/resources/runtime/services.cfg [hunk 2]
```diff
@@ -14,8 +14,9 @@ threadpool:safeCall=10
...
+startlevel:30=persistence:restore,automation:scriptEngineFactories
+startlevel:40=dsl:rules,managed:rule,rules:refresh,rules:dslprovider,automation:scriptEngineFactories
```
Comment (kaikreuzer) [inline distributions/openhab/src/main/resources/runtime/services.cfg:19]: Why does `automation:scriptEngineFactories` appear here again? Imho, every identifier should only occur at exactly one start level definition.

File: distributions/openhab/src/main/resources/runtime/services.cfg [hunk 3]
```diff
@@ -14,8 +14,9 @@ threadpool:safeCall=10
...
 startlevel:40=dsl:rules,managed:rule,rules:refresh,rules:dslprovider
```
Comment (kaikreuzer) [inline distributions/openhab/src/main/resources/runtime/services.cfg:20]: I guess this line should now be removed.

**Review by kaikreuzer — CHANGES_REQUESTED** [review]

**Comment by J-N-K** [comment]
All done.

**Review by kaikreuzer — APPROVED** [review]
Thanks!

**Pull request merged**